autoria.py

```diff
@@ -0,0 +1,242 @@
+    """Get available car marks from auto.ria.com.
+
+    Args:
+        category - integer with needed category id.
+
+    Returns:
+        JSON-formatted list of pairs ''name: value'' like:
+        [
+            {
+            "name": "Acura",
+            "value": 98
+            },
+            ...
+        ]
+    """
+    url = 'http://api.auto.ria.com/categories/{}/marks' \
+          .format(str(category))
+    return make_request(url)
+
+
+def get_models(category: int, mark: int) -> json:
+    """Get available models for selected mark from auto.ria.com.
+
+    Args:
+        category - integer with needed category id.
+        mark - integer with needed mark id
+            (since we're getting models of the mark)
+
+    Returns:
+        JSON-formatted list of pairs ''name: value'' like:
+        [
+            {
+            "name": "CL",
+            "value": 953
+            },
+            ...
+        ]
+    """
+    url = 'http://api.auto.ria.com/categories/{}/marks/{}/models' \
+          .format(str(category), str(mark))
+    return make_request(url)
+
+
+def get_states() -> json:
+    """Get available states from auto.ria.com.
+
+    Args aren't needed because the url is fixed.
+
+    Returns:
+        JSON-formatted list of pair ''name: value'' like:
+        [
+            {
+            "name": "Винницкая",
+            "value": 1
+            },
+            ...
+        ]
+    """
+    url = 'http://api.auto.ria.com/states'
+    return make_request(url)
+
+
+def select_item(item_to_select: str, items_list: json) -> int:
+    """Select vehicle type, bodystyle, mark, model from the given list.
+
+    This function is intended to convert human-readable search parameter,
+    for instance, ''Винница'' into API-understandable state identifier,
+    for the given example it would be 1.
+
+    Args:
+        item_to_select - string, could be not 100% accurate as it is in the
+            auto.ria.ua lists, e.g. value ''Харьков'' is acceptable,
+            because the parameter is wildcarted in comprasion like
+            ''*Харьков*'', the function will find suitable name
+            ''Харьковская'' and will return its id.
+        items_list - JSON-formatted list of pairs ''name: value'',
+                obtained from one of the ''get_'' functions.
+
+    Returns:
+        integer with needed item (category, bodystyle, mark etc.) identifyer.
+    """
+    if item_to_select is not None and items_list is not None:
+        for item in items_list:
+            if fnmatch(item['name'], '*' + item_to_select + '*'):
+                return item['value']
+    else:
+        return None
+
+
+def get_average(category: str, mark: str, model: str, bodystyle: str = None,
+                years_start_end: list = None, state: str = None) -> json:
+    """Compose parameters for GET request and send it.
+
+    Get average price using search parameters.
+
+    Args:
+        category - (required) string with vehicle type,
```

Comment:
It is not necessary to put required / optional too - this is clear from default values for function parameters.
I may be helpful to use "optional" though to turn the reader attention that parameter is optional, but required is not needed for sure, we assume that all function parameters are required "by default", this is how functions work (we should explicitly specify the optional parameter value, but the parameter itself still exists).